great_expectations/datasource/fluent/pandas_datasource.py

```diff
@@ -513,8 +523,19 @@ def _add_asset(
 
         asset_names: Set[str] = self.get_asset_names()
 
+        in_cloud_context: bool = False
+        if self._data_context:
+            in_cloud_context = self._data_context._datasource_store.cloud_mode
+
         if asset_name == DEFAULT_PANDAS_DATA_ASSET_NAME:
-            if asset_name in asset_names:
+            if in_cloud_context:
+                # In cloud mode, we need to generate a unique name for the asset so that it gets persisted
+                asset_name = f"{asset.type}-{_short_id()}"
+                logger.info(
+                    f"Generating unique name for '{DEFAULT_PANDAS_DATA_ASSET_NAME}' asset '{asset_name}'"
+                )
+                asset.name = asset_name
+            elif asset_name in asset_names:
                 self.delete_asset(asset_name=asset_name)
```

Comment:
We're just assuming no collisions right? Was thinking this should be an `if` but no way this will ever be a material issue